llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -965,6 +965,40 @@ static void updateDefinedRegisters(MachineInstr &MI, LiveRegUnits &Units,
       Units.addReg(MOP.getReg());
 }
 
+/// Find the DBG_INSTR_REF instruction that references the \p InstrNum
+static std::optional<MachineInstr *> findDebugInstrRef(MachineBasicBlock *MBB,
+                                                       unsigned InstrNum) {
+
+  for (auto &MI : *MBB) {
+    if (MI.isDebugRef())
+      if (MI.getOperand(2).getInstrRefInstrIndex() == InstrNum)
+        return &MI;
+  }
+  return std::nullopt;
+}
+
+/// Set the correct debug-instr-number and the operand index in case of a merge.
+static void setDebugInstrNum(MachineBasicBlock::iterator OriginalInstr,
+                             MachineInstrBuilder &MIB, unsigned InstrNumToFind,
+                             unsigned InstrNumToSet, MachineBasicBlock *MBB) {
+
+  auto InstrRefMI = findDebugInstrRef(MBB, InstrNumToFind);
+  if (InstrRefMI) {
+    auto *MI = *InstrRefMI;
+    MI->getOperand(2).setInstrRefInstrIndex(InstrNumToSet);
+    // Set the Instruction Reference Op Index to be the same as the operand of
```

Comment:
Then this comment can move to the call site, where it makes more sense in context.

Reply:
I updated the comment for the function and changed the name to be `findDebugInstrRefAndUpdateInstrNumAndOpIndex` but I don't think this comment should be in the call site, because the function does more than just set the Op Index.